profiling-ffi/src/profiles.rs

```diff
@@ -719,41 +709,18 @@ unsafe fn add_upscaling_rule(
     )
 }
 
-#[repr(C)]
-pub struct EncodedProfile {
-    start: Timespec,
-    end: Timespec,
-    buffer: ddcommon_ffi::Vec<u8>,
-    endpoints_stats: Box<ProfiledEndpointsStats>,
-}
```

Comment:
Hmmm removing this leaves us in a weird in-between state -- I did find it useful to have the `start`/`end` here because in some cases I relied on libdatadog's defaults.

Providing start/end time is currently optional on a number of our APIs, so I think if we remove them here I think it's worth going ahead and also make them non-optional on the other APIs -- particularly in `ddog_prof_Profile_serialize` (make non-optional) and possibly in `ddog_prof_Profile_reset`/`ddog_prof_Profile_new` (remove them?).

Reply:
I'd (mildly) rather do that on another PR so we can figure out the API we want, and when to use the current time vs requiring a time be passed in.

Reply:
Hmmm my suggestion is to: 

* Remove `start_time` from `ddog_prof_Profile_new` [GitHub search showing that dd-trace-py/ddprof/dd-trace-dotnet/dd-trace-rb all pass in a NULL pointer for that argument](https://github.com/search?q=org%3ADataDog%20ddog_prof_Profile_new&type=code)
* Remove `start_time` from `ddog_prof_Profile_reset` [GitHub search showing that dd-trace-py/ddprof pass in a NULL; Ruby passes in non-null but I'll change it](https://github.com/search?q=org%3ADataDog+ddog_prof_Profile_reset&type=code)
* Leave `ddog_prof_Profile_serialize` untouched

Reply:
Done